### PR TITLE
Adding a PDA module for medics to get the attention of nearby people

### PR DESCRIPTION
**Pull request by Kaan-NN**
[QOL]

## About the PR 
I added a new pda module for the MD and everyone with the medical PDA.
This inherits from flashlight and therefore retains the default flashlight settings.
It works in similar fashion as the segway, but should be both auditory and visually distinct from it.
It plays the [https://github.com/goonstation/goonstation/blob/master/sound/effects/manta_alarm.ogg](url), as it currently is not used for commonly encountered stuff, and creates a slightly larger light then the segway.
It adds an ability button to the hud that triggers the effect with a 90 second cooldown (this can and should be changed if players decide to spam it to be annoying)

https://user-images.githubusercontent.com/68946148/168671158-16ae78ae-e0c7-4b5c-a040-42266a25a834.mp4

https://user-images.githubusercontent.com/68946148/168671165-84ce905e-4101-4110-94bb-8cddfcf21893.mp4

UI_preview
icon_preview

## Why's this needed? 
People that are heavily injured are only focused on getting to medbay no matter what, frequently ignoring any medical personnel that tries to help them, making it an overall annoying experience for both people involved. Since it works similar to the segway it could also make for a paramedic gimmick of running in clearing halls to make room for someone being dragged to medbay or doctors running through a crowded area towards the patient and could therefore be useful for RP.

## Changelog

```changelog
(u)Kaan
(+)Medical personnel now starts with a PDA module that sounds a siren to announce their presence.
```

### CODE REVIEW

**Comment by Kaan-NN** [comment]
> 

image
updated the icon to be a bit smaller

**Comment by ZeWaka** [comment]
Definitely too loud

**Comment by Sbmhawk** [comment]
And too bright imo

**Comment by Kaan-NN** [comment]
I adjusted sound and brightness I hope the audio is fine this time I used the doors and owls as a reference since OBS does some weird audio stuff I havent figured out yet

https://user-images.githubusercontent.com/68946148/168797892-2bc63e79-9318-4d7d-9c0e-2546082a178d.mp4

File: _std/__build.dm [hunk 1]
```diff
@@ -30,17 +30,17 @@ o+`        `-` ``..-:yooos-..----------..`
 
 //////////// OPTIONS TO GO FAST
 
-//#define IM_REALLY_IN_A_FUCKING_HURRY_HERE 1  // Skip setup for atmos, Z5, don't show changelogs, skip pregame lobby
```
Comment (TobleroneSwordfish) [inline _std/__build.dm:33]: Don't commit __build.dm

File: _std/compatibility.dme [hunk 2]
```diff
@@ -10,3 +10,12 @@
 #else
 #define BYOND_VERSION_OK
 #endif
+// BEGIN_INTERNALS
+// END_INTERNALS
+// BEGIN_FILE_DIR
+#define FILE_DIR .
```
Comment (TobleroneSwordfish) [inline _std/compatibility.dme:16]: don't think this is meant to be committed either

File: code/obj/item/device/pda2/modules.dm [hunk 3]
```diff
@@ -321,3 +322,64 @@
 		var/obj/item/device/pda_module/alert/J = the_item
 		if (J.host)
 			J.send_alert()
+
+/obj/item/device/pda_module/flashlight/yell_medic
+	name = "medic siren module"
+	desc = "A PDA module that lets you announce your presence as medical personnel to your surroundings."
+	icon_state = "pdamod_medic"
+	abilities = list(/obj/ability_button/pda_flashlight_toggle, /obj/ability_button/yell_medic)
+	var/datum/light/MedicLight
```
Comment (TobleroneSwordfish) [inline code/obj/item/device/pda2/modules.dm:331]: Consider changing this to a simple light, as robust lights look very laggy when attached to a moving object (and the player is always going to be moving if they're pressing this)

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
Overall I like the idea, one of the most frustrating things about playing medic

**Comment by Kaan-NN** [comment]
Now using simple light with colors that should be decently visible
https://user-images.githubusercontent.com/68946148/169117460-940c86e0-f8ad-4a42-aaf3-6007d97ac1da.mp4

File: code/obj/item/device/pda2/modules.dm [hunk 4]
```diff
@@ -321,3 +322,56 @@
...
+
+	proc/toggle_yell()
+		if(!src.host)
+			boutput(usr, "<span class='alert'>No PDA detected.")
+			return
+		if(ON_COOLDOWN(src, "toggle_yell", 90 SECONDS))
+			boutput(usr, "<span class='alert'>[src] is still on cooldown mode!</span>")
+			return
+		tell_medic()
+		ColorChangeCycles = 6
```
Comment (Tarmunora) [inline code/obj/item/device/pda2/modules.dm:351]: Why is this initialized here and not when defined

File: code/obj/item/device/pda2/modules.dm [hunk 5]
```diff
@@ -321,3 +322,56 @@
...
+			sleep(0.4 SECONDS)
+			usr.remove_sm_light("MedLight")
+			ColorChangeCycles--
+
+	proc/tell_medic()
+		switch(rand(100))
+			if(0 to 97)
+				Messages = list("Doctor, this way!", "Medical personnel coming through", "[usr] is a medical professional", "[usr] can provide medical assistance", "Medical support here", "Doctor here", "[usr] is a doctor", "MEDIC HERE", "Don´t run from [usr] [usr.get_pronouns().subjective] can patch you up", "[usr] will heal you", "[usr] will patch you up", "Get healed by [usr]", "[usr] is a doctor [usr.get_pronouns().subjective] can heal you", "[usr] is here to heal you", "Medic here to patch you up")
+			if(98 to 100)
+				Messages = list("Boss, thats a serious injury. It won´t heal on its own. You´ll need to treat it with first aid.", "Ready to heal! Dummkopfs", "Stop running you half dead moron [usr] will heal you because you aren´t gonna make it to the crater that once WAS mebay and [usr] stockpiled on every single med known to mankind only for this very moment")
```
Comment (Tarmunora) [inline code/obj/item/device/pda2/modules.dm:367]: give this another pass for punctuation and spelling

**Comment by Conchuckter** [comment]
One thing that would make this better is if this sent a PDA notification to the medical department like the security alert module.

**Comment by flrsh** [comment]
I agree with the motivation behind this PR, but I'm not sure this effect would actually compel people to, like, stop running away when you're trying to heal them or whatever. If anything, if I were a new player, I might want to run away from "person with bright flashing light and sound effect" even more.

**Comment by Kaan-NN** [comment]
> I agree with the motivation behind this PR, but I'm not sure this effect would actually compel people to, like, stop running away when you're trying to heal them or whatever. If anything, if I were a new player, I might want to run away from "person with bright flashing light and sound effect" even more.

As someone on the forum stated people should get used to it over time. Especially during the honeymoon phase where doctors will likely spam it without end people will primarily see it used by medical staff and since it is imo distinct enough from other types of sirens currently used in game people should come to associate them them doctors quite quickly.

There was also someone that suggested to make an AI hologram styled cross appear over the medic but I tend overlook these so I disregarded that for now. If that is actually preferred Id be up to change that.

**Comment by mordent-goonstation** [comment]
Not personally a fan, this seems comically obnoxious. If someone is able to evade you to the point of you needing to turn this on to get their attention rather than being able to hear you say "I'm a doctor!" then they aren't hurt badly enough for it to matter. If they aren't able to evade you, you can say the thing and if they don't want to be treated then they'll still crawl away.

I don't think this will actually solve the problem it is trying to solve.

**Comment by Tarmunora** [comment]
All things considered I don't see this going in. Unless another dev has strong feelings one way or another, closing.

**Comment by Kaan-NN** [comment]
> All things considered I don't see this going in. Unless another dev has strong feelings one way or another, closing.

Since this was my first PR could you provide me with some information on why you dislike the idea, so that I can eventually rework the entire idea or at least know what to improve on for future PRs?

**Comment by Tarmunora** [comment]
🤷 didn't vibe.

Also see mordent's response, because he sums it up pretty well

**Pull request closed without merging**